Dockerfile

```diff
@@ -77,6 +77,7 @@ ARG RUSTFLAGS
 ARG LINKER=mold
 
 COPY --from=planner /qdrant/recipe.json recipe.json
+COPY . .
```

Comment:
Removing `.git` from `.dockerignore` and using `COPY . .` in `Dockerfile` puts the entire `.git` directory into a docker layer/context and thus it could render the docker caching ineffective.

Instead, I suggest to pass git revision to the docker either via an environment variable using a single file. E.g. [like this](https://www.github.com/moby/moby/issues/14191#issuecomment-242718433).

Reply:
> puts the entire .git directory into a docker layer/context and thus it could render the docker caching ineffective.

Great point. I didn't consider this before. Thanks!

Reply:
@KShivendu I believe we have suggested this quite a few times now.

Also, I'm afraid the changes in the current PR break in the same way the last PR did, which is why it was reverted.

If not, please feel free to elaborate!